iit/model_pairs/base_model_pair.py

```diff
@@ -177,7 +197,7 @@ def get_IIT_loss_over_batch(
         hl_output, ll_output = self.do_intervention(base_input, ablation_input, hl_node)
         label_idx = self.get_label_idxs()
         # IIT loss is only computed on the tokens we care about
-        loss = loss_fn(ll_output[label_idx.as_index], hl_output[label_idx.as_index])
+        loss = loss_fn(ll_output[label_idx.as_index].to(hl_output.device), hl_output[label_idx.as_index])
```

Comment:
We should probably just raise if dataset, hl_model and ll_model aren't on the same device during init/starting training. This usually just hides the main problem and makes it harder to find bugs.

Reply:
Makes sense, I'll add an assert to the beginning of train() and remove all of these.